### PR TITLE
Add relative to other tests 'run' mark 'after' and 'before' kwargs

### LINKED ISSUE #65: The module relative order don't working???
this module don't working?
```python
import pytest
@pytest.mark.run(after='test_second')
def test_third():
    assert True

def test_second():
    assert True

@pytest.mark.run(before='test_second')
def test_first():
    assert True
```

**The order don't change all ???**
<pre>
E:\QT-learn>pytest -svv
============================================================================= test session starts =============================================================================
platform win32 -- Python 3.7.4, pytest-5.4.3, py-1.9.0, pluggy-0.13.1 -- c:\users\jekoie\appdata\local\programs\python\python37-32\python.exe
cachedir: .pytest_cache
rootdir: E:\QT-learn, inifile: pytest.ini
plugins: ordering-0.6
collected 3 items                                                                                                                                                              

test_qt.py::test_third PASSED
test_qt.py::test_second PASSED
test_qt.py::test_first PASSED

============================================================================== 3 passed in 0.04s ==============================================================================

E:\QT-learn>

</pre>

### PR DESCRIPTION
**Pull request by mondeja**
Closes #65 

As was reported, [relative to other tests module ordering](https://pytest-ordering.readthedocs.io/en/develop/#relative-to-other-tests) is not working because is not implemented.

### CODE REVIEW

File: pytest_ordering/__init__.py [hunk 1]
```diff
@@ -62,6 +63,20 @@ def pytest_collection_modifyitems(session, config, items):
 
         if mark:
             order = mark.kwargs.get('order')
+            if order is None:
+                before = mark.kwargs.get('before')
+                if before:
+                    if not before in before_items:
+                        before_items[before] = []
+                    before_items[before].append(item)
```
Comment (mrbean-bremen) [inline pytest_ordering/__init__.py]: this is usally written as:
```py
before_items.setdefault(before, []).append(item)
```
(same as for order)

File: pytest_ordering/__init__.py [hunk 2]
```diff
@@ -80,3 +95,22 @@ def pytest_collection_modifyitems(session, config, items):
...
+    def _get_item_index_by_name(item_name):
+        index = None
+        for i, item in enumerate(items):
+            if getattr(item, "name") == item_name:
+                index = i
+                break
+        return index
+        
+    for before_item_relative, _before_items in before_items.items():
+        index = _get_item_index_by_name(before_item_relative)
```
Comment (mrbean-bremen) [inline pytest_ordering/__init__.py:104]: Note that for invalid test names in `before` or `after` the test is just skipped - would probably be better to issue a warning and ignore the marker

**Review by mrbean-bremen — COMMENTED** [review]
I actually prepared a PR for the same feature, but had trouble to get it running in the CI (a compatibility issue with pytest-cov) - so you beat me to it :)
It would also make sense to update the documentation (docs/index.rst) to move the related part out of the "Aspirational" chapter.

File: pytest_ordering/__init__.py (hunk 2 above)
Comment (mondeja) [reply pytest_ordering/__init__.py:104]: Fixed now. Also, tests for warnings added.

File: pytest_ordering/__init__.py [hunk 3]
```diff
@@ -80,3 +91,60 @@ def pytest_collection_modifyitems(session, config, items):
...
+        if index is not None:
+            for before_item in _before_items:
+                items.insert(index, before_item)
+        else:
+            if len(_before_items) == 1:
+                message_schema = "%s, indicated at parameter before of" \
+                               + " %s, doesn't exists"
```
Comment (mrbean-bremen) [inline pytest_ordering/__init__.py]: Nitpicking: "doesn't exist"

File: pytest_ordering/__init__.py [hunk 4]
```diff
@@ -80,3 +91,60 @@ def pytest_collection_modifyitems(session, config, items):
...
+                test_names = ""
+                for i, before_item in enumerate(_before_items):
+                    test_names += before_item.name
+                    if i < len(_before_items) - 2:
+                        test_names += ", "
+                    elif i == len(_before_items) - 2:
+                        test_names += " and "
+                message = message_schema % (before_item_relative, test_names)
+            warnings.warn(message, SyntaxWarning)
```
Comment (mrbean-bremen) [inline pytest_ordering/__init__.py:125]: Nice!

**Review by mrbean-bremen — APPROVED** [review]
Looks good to me, thanks! 
@ftobia - the [Travis builds](https://travis-ci.org/github/ftobia/pytest-ordering) are currently failing due to an incompatibility with the latest version of pytest-cov - I'm looking into this. But they are also not visible on the PR; I had a [similar issue](https://travis-ci.community/t/github-ci-notifiactions-from-travis-stopped-working/9238) recently, still unresolved - maybe you can check the stettings.

**Comment by mrbean-bremen** [comment]
@mondeja - I didn't notice at the time, but there is already a PR (#37) that implements this. I integrated that in [my fork](https://github.com/mrbean-bremen/pytest-order) (I was checking the oldest PRs first), so yours didn't get integrated, although I had planned to do so first...

**Comment by mondeja** [comment]
> so yours didn't get integrated, although I had planned to do so first...

Don't worry about that.

**Pull request closed without merging**